connectors/sources/sharepoint_online.py

```diff
@@ -1147,6 +1147,7 @@ def __init__(self, configuration):
 
         self._client = None
         self.site_group_cache = {}
+        self._temporary_cursor = None
```

Comment:
I don't think I understand what's changed here.
This doesn't implement checkpointing for SPO, AFAICT. But this is just storing the cursor info we used to have in `self._sync_cursor` instead in `self._temporary_cursor`, and then overwriting `self._sync_cursor` with `self._temporary_cursor` at the end. Why keep it in to variables? As long as the intermediate `self._sync_cursor` isn't persisted to ES, it's effectively "temporary" because it's just in-memory for this instance of the DataSource, right?

Reply:
That is the thing: to make checkpoints work, connector needs to be able to write its state to some variable. Since we use the same `connector.sync_cursor`, connector will store checkpoints in `self._sync_cursor` variable (see github implementation).

Then framework checks this variable from time to time, and if it changed, it's saved into the job state (not connector state).

Now thinking of it, this change is not necessary because the connector state is updated at the end of successful sync, but all of it is just highly confusing.

This is the main reason I wanted checkpoints separated - our current way of arranging things (different job types, incremental cursors) just amplify the complexity of checkpointing a lot. To have something solid we'd need a good revamp of this logic, now it feels it works but is very fragile